test/framework/workload.go

```diff
@@ -54,15 +61,96 @@ func (w *WorkloadCluster) DeleteClusterWithKubectl() {
 	w.StopIfFailed()
 }
 
-func (w *WorkloadCluster) applyClusterManifest(ctx context.Context) {
-	if err := w.KubectlClient.ApplyManifest(ctx, w.ManagementClusterKubeconfigFile(), w.ClusterConfigLocation); err != nil {
-		w.T.Fatalf("Failed to apply workload cluster config: %s", err)
+// CreateClusterWithAPIGitOps uses gitops to create a cluster with the API.
+func (w *WorkloadCluster) CreateClusterWithAPIGitOps() {
+	ctx := context.Background()
+	w.createWithAPIGitOps(ctx)
+	w.waitForKubeconfig(ctx)
+	w.ValidateClusterState()
+	w.T.Logf("Successfully created version controlled cluster configuration")
+}
+
+func (w *WorkloadCluster) createWithAPIGitOps(ctx context.Context) {
+	w.ManagementCluster.initGit(ctx)
+	// Pull remote config using managment cluster
+	err := w.ManagementCluster.pullRemoteConfig(ctx)
+	if err != nil {
+		w.T.Errorf("pulling remote configuration: %v", err)
+	}
+
+	w.T.Log("Creating local cluster config file in git repo for creation")
+	// Marshall w.ClusterConfig and write it to the repo path
+	w.buildClusterConfigFileInManagementClusterForGit()
+
+	if err := w.addConfigChangesInManagementClusterToGit(ctx); err != nil {
+		w.T.Errorf("Error pushing local changes to remote git repo: %v", err)
+	}
+}
+
+// UpgradeClusterWithAPIGitOps uses gitops to upgrade a cluster with the API.
+func (w *WorkloadCluster) UpgradeClusterWithAPIGitOps(clusterOpts ...ClusterE2ETestOpt) {
```

Comment:
Why is this different than create? isn't it the same, just writing a file to disk, committing and pushing? The one who knows which operation is a create a which one is an upgrade is the test itself, since that's the entity who orchestrates the operations.